oauthenticator/github.py

```diff
@@ -113,6 +114,21 @@ def _github_client_secret_changed(self, name, old, new):
         config=True, help="Automatically allow members of selected organizations"
     )
 
+    fetch_teams = Bool(
+        False,
+        help="""
+        Fetch list of teams user is in and put it in auth_state.
+
+        'teams' will be a key in auth_state that has the same structure listed
+        in https://docs.github.com/en/rest/reference/teams#list-teams-for-the-authenticated-user.
```

Comment:
Instead of saying it `will be a key in auth_state` I think it would be relevant to clarify that this is put in `auth_state` during authentication, making it available to a `Authenticator.post_auth_hook` but not for a Spawner hook of some kind, which would also require `enable_auth_state` to be set to True.

I'm mainly concerned that people assume it will be available, but then it isn't.

Related links:
- Docs on `Authenticator.post_auth_hook`: https://jupyterhub.readthedocs.io/en/stable/api/auth.html#jupyterhub.auth.Authenticator.post_auth_hook
- Discourse topic on using auth state etc: https://discourse.jupyter.org/t/tailoring-spawn-options-and-server-configuration-to-certain-users/8449

Reply:
> I'm mainly concerned that people assume it will be available, but then it isn't.

I've clarified in the string that this works only if auth_state is enabled. I'd like to not repeat more details about auth_state here though, users can look through docs about auth_state for that.

Reply:
It will work for the Authenticator class post_auth_hook though, it just wont be persisted to be around in auth state for later use by a spawner.

I fully agree on avoiding repetition, linking out to relevant socs to explain details is great!

Reply:
@consideRatio we don't actually have a 'topic' page about auth_state, so I'm not sure what to link out to from here. I'm also running out of energy unfortunately, and want to avoid shaving too many yaks! So instead of trying to figure out best way to tell people what auth_state is here, can I just let things be as they are, and just make sure that people know this only puts things into auth_state if it is enabled?

Reply:
Thank you Yuvi for your work to think about this! I've opened https://github.com/jupyterhub/oauthenticator/pull/498/files#r860745304 to make me feel happy about this documentation detail!